app/src/main/java/ch/epfl/sdp/peakar/points/ComputePOIPoints.java

```diff
@@ -37,17 +40,51 @@
      * @param context Context of activity
      */
     public ComputePOIPoints(Context context){
+        ctx = context;
         POIPoints = new ArrayList<>();
         userPoint = UserPoint.getInstance(context);
         userPoint.update();
         getPOIs(userPoint);
     }
 
+
     /**
-     * Gets the POIs for the userPoint
-     * @param userPoint location of the user
+     * Retrieves list of surrounding POIs either from cache
+     * or from provider. If the cached data corresponds to
+     * the user location, no download is made and the POIs
+     * are retrieved from cached file
+     * @param userPoint user location
      */
     private static void getPOIs(UserPoint userPoint){
+        //Retrieve cache instance
+        POICache poiCache = POICache.getInstance();
+        //Check if the user is in the cached BB. If yes
+        //load data from cache / if not download data
```

Comment:
```suggestion
```

Thinking about the review we got from TA to remove these type of comments. I think it is well explained in the javadoc

Reply:
You are right